pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SortOperator.java

```diff
@@ -164,6 +162,9 @@ private void consumeInputBlocks() {
               _rows.addAll(container.subList(0, _numRowsToKeep - numRows));
               LOGGER.debug("Early terminate at SortOperator - operatorId={}, opChainId={}", _operatorId,
                   _context.getId());
+              setEarlyTerminate();
```

Comment:
This is the only place where early terminate can be set --> and only acquire one extra block b/c the downstream should guarantee after the early terminate is set. the next block must've been a metadata block.